server.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/chromium/hstspreload"
+	"github.com/chromium/hstspreload/chromiumpreload"
+)
+
+func main() {
+	staticHandler := http.FileServer(http.Dir("files"))
+	http.Handle("/", staticHandler)
+	http.Handle("/static/", staticHandler)
+
+	http.HandleFunc("/robots.txt", http.NotFound)
+	http.HandleFunc("/favicon.ico", http.NotFound)
+
+	http.HandleFunc("/checkdomain/", checkdomain)
+	http.HandleFunc("/status/", status)
+
+	http.HandleFunc("/submit/", submit)
+	http.HandleFunc("/pending", pending)
+	http.HandleFunc("/update", update)
+
+	http.ListenAndServe(":8080", nil)
+}
+
+func checkdomain(w http.ResponseWriter, r *http.Request) {
+	domain := r.URL.Path[len("/checkdomain/"):]
+
+	_, issues := hstspreload.PreloadableDomain(domain)
+
+	b, err := json.MarshalIndent(issues, "", "  ")
+	if err != nil {
+		http.Error(w, "Internal error: could not encode JSON.\n", http.StatusInternalServerError)
+	} else {
+		fmt.Fprintf(w, "%s\n", b)
+	}
+}
+
+// writeJSONOrBust should only be called if nothing has been written yet.
+func writeJSONOrBust(w http.ResponseWriter, v interface{}) {
+	w.Header().Set("Content-type", "text/css; charset=utf-8")
+
+	b, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not format JSON. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+
+	fmt.Fprintf(w, "%s\n", b)
+	return
+}
+
+func status(w http.ResponseWriter, r *http.Request) {
```

Comment:
Go encourages concise names. For the time being, I want to name the handler functions the same as the API paths they handle.